core/controllers/base_test.py

```diff
@@ -1546,9 +1546,16 @@ def test_schema_keys_exactly_match_with_url_path_elements(self) -> None:
                 continue
 
             regex_pattern = r'<.*?>'
-            url_path_elements = [
-                keyword[1:-1] for keyword in re.findall(
-                    regex_pattern, route.name)]
+            url_path_elements = []
+            for keyword in re.findall(regex_pattern, route.name):
+                # In some cases, url_path_arguments are defined with specific
+                # acceptable values, e.g: /<asset_type:(image|audio|thumbnail)>.
+                # So, to separate the argument name from acceptable values, we
+                # have used ':' delimiter's index.
```

Comment:
I think it's worth noting that, in the case where ':' doesn't exist, the url_argument_delimiter_index is -1 ... which, by chance, is actually what you want it to be since you're trimming the last character.

But that's not obvious conceptually. So I suggest you actually parse out the thing between `<` and `>`, and _then_ trim the ':' if it exists. Otherwise this is hard to maintain and reason through intuitively.

Reply:
re.findall() can return a list of strings iff it found something. 

> So I suggest you actually parse out the thing between < and >, and then trim the ':' if it exists.

a keyword can only exist if we have something between <>. Because if URL does not contain `<`and `>` then re.findall will return an empty list. But is there a case we have `some_url/<>`?, because in that case only, keyword exist with an empty string?

Reply:
No, you wouldn't have an empty string here.

Basically, the previous code took out the part between `<` and `>`. I am suggesting continuing to do that, and then finding a ':' in that string if needed.